index.js

```diff
@@ -225,21 +228,88 @@ module.exports = {
         // Save the new sha
         testIndexSha = result.sha;
 
-        // The list of scripts has changed or the list of amd modules has changed, either way rebuld the
-        // start script
-        this.startScriptBuilder(result);
       }.bind(this)).catch(function () {
         // If there is no tests/index.html, the function will reject
         return;
       });
 
       return RSVP.all([indexPromise, testIndexPromise]);
+      //return RSVP.all([indexPromise]);
     }.bind(this));
   },
 
   indexBuilder: function (config) {
+    //----------------------------------------------------------------------------
+    //STEPS:
+    //1 - handle the configFile if defined
+    //1 - get script names from index.html
+    //2 - cook the amd-start script as a string
+    //3 - if inline === true : inline the scripts
+    //  - if inline === false : add script tags to the amd-start/amd-config files
+    //----------------------------------------------------------------------------
     
-    // Load the index file    
+    // Full path to the index file    
+    var indexPath = path.join(config.directory, config.indexFile);
+
+    //1 - deal with the amd config file if defined
+    var amdConfigInfo = this.handleAmdConfig(config);
+
+    var self = this;
+    //2 - get the scripts from index.html
+    return this.getScriptsFromIndex(indexPath)
+      .then(function(scripts){
+        config.scriptsAsString = scripts;
+        //3- create the start script
+        var amdStartScriptInfo = self.startScriptBuilder(config);  
+        //if we are inlining...
+        if(config.inline){
+          config.amdConfig = amdConfigInfo.amdConfig;
+          config.amdStart = amdStartScriptInfo.amdStart;
+          return self.inlineScripts(config);
+        }else{
+          config.amdConfigFileName = amdConfigInfo.amdConfigFileName;
+          config.amdStartFileName = amdStartScriptInfo.amdStartFileName;
+          return self.replaceScripts(config);
+        }
+      })
+      .then(function(result){
+        return result;
```

Comment:
As a general code style, we used .bind(this) in the rest of the code. I would prefer to continue to use the same pattern.